sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -232,18 +229,13 @@ class UnivocityParser(
         new RuntimeException("Malformed CSV record"))
     }
 
-    var checkedTokens = tokens
     var badRecordException: Option[Throwable] = None
```

Comment:
How about:
```scala
    var badRecordException: Option[Throwable] = if (tokens.length != parsedSchema.length) {
      // If the number of tokens doesn't match the schema, we should treat it as a malformed record.
      Some(new RuntimeException("Malformed CSV record"))
    } else None
```

Reply:
Yup, I did :)

Reply:
@HyukjinKwon You address comments before I post them ;-)